### ISSUE TITLE
Link-name should allow titles

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#2459]
In axe-core 4.0 we changed how has-visible-text behaves. It now only looks at content, not at accessible name props for things. Because of that, the following code snippet now fails the link-name rule:

```html
<a href="http://www.w3.org/WAI" title="Web Accessibility Initiative">
	<img src="/test-assets/shared/w3c-logo.png" alt="" />
</a>
```

The reason for this is that the `non-empty-title` check is not used in the `link-name` rule (as as it is for things like `button-name`). I vaguely remember we did that because we didn't want to encourage the use of the title attribute. I think support has improved since that time, so I think it's save to add it.

**Comment by DaveTryon** [comment]
@WilcoFiers, any thoughts on when a release might be published with this? We're working towards using axe-core 4.0.1 in Accessibility Insights and are trying to determine our best options. Thanks!

**Comment by padmavemulapati** [comment]
Link-name rule should pass if title present fail otherwise
![image](https://user-images.githubusercontent.com/47409579/93730759-a6a3ef00-fbe7-11ea-801c-eba88b861603.png)

**Comment by straker** [comment]
No docs required
